src/utils/decorators/input-validation/input-validation.js

```diff
@@ -167,6 +171,93 @@ const InputValidation = ComposedComponent => class Component extends ComposedCom
     info: PropTypes.array
   });
 
+  componentDidUpdate(prevProps) {
+    if (super.componentDidUpdate) { super.componentDidUpdate(prevProps); }
+
+    if (!this.state.valid || this.state.warning || this.state.info) {
+      this.positionElements();
```

Comment:
do we need to run this on every update? can we try and minimise how many times it needs to be called somehow?